application/src/main/java/org/togetherjava/tjbot/commands/filesharing/FileSharingMessageListener.java

```diff
@@ -0,0 +1,183 @@
+            "fxml", "css", "c", "h", "cpp", "py", "yml");
+
+    private final Predicate<String> isStagingChannelName;
+    private final Predicate<String> isOverviewChannelName;
+
+
+    public FileSharingMessageListener(@NotNull Config config) {
+        super(Pattern.compile(".*"));
+
+        gistApiKey = config.getGistApiKey();
+        isStagingChannelName = Pattern.compile(config.getHelpSystem().getStagingChannelPattern())
+            .asMatchPredicate();
+        isOverviewChannelName = Pattern.compile(config.getHelpSystem().getOverviewChannelPattern())
+            .asMatchPredicate();
+    }
+
+    @Override
+    public void onMessageReceived(@NotNull MessageReceivedEvent event) {
+        User author = event.getAuthor();
+        if (author.isBot() || event.isWebhookMessage()) {
+            return;
+        }
+
+        if (!isHelpThread(event)) {
+            return;
+        }
+
+        List<Message.Attachment> attachments =
+                event.getMessage().getAttachments().stream().filter(attachment -> {
+                    String extension = attachment.getFileExtension();
+                    if (extension == null) {
+                        return false;
+                    }
+                    return extensionFilter.contains(extension);
+                }).toList();
+
+        processAttachments(event, attachments);
+    }
+
+    private void processAttachments(@NotNull MessageReceivedEvent event,
+            @NotNull List<Message.Attachment> attachments) {
+        Map<String, GistFile> filesAsJson = new HashMap<>();
+
+        for (Message.Attachment attachment : attachments) {
+            attachment.retrieveInputStream()
+                .thenApply(this::readAttachment)
+                .thenAccept(content -> filesAsJson.put(createFileName(attachment),
+                        new GistFile(content)))
+                .join();
+        }
+
+        GistFiles files = new GistFiles(filesAsJson);
+        GistRequest request = new GistRequest(event.getAuthor().getName(), false, files);
+        String url = uploadToGist(request);
+        sendResponse(event, url);
+    }
+
+    private @NotNull String readAttachment(@NotNull InputStream stream) {
+        try {
+            return new String(stream.readAllBytes(), StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private @NotNull String createFileName(@NotNull Message.Attachment attachment) {
+        String fileName = attachment.getFileName();
+        String fileExtension = attachment.getFileExtension();
+
+        if (fileExtension == null || fileExtension.equals("txt")) {
+            fileExtension = "java";
+        } else if (fileExtension.equals("fxml")) {
+            fileExtension = "xml";
+        }
+
+        fileName = fileName.substring(0, fileName.lastIndexOf(".")) + "." + fileExtension;
```

Comment:
this still has the problem of crashing if `fileName` has no `.` in it (`lastIndexOf` retuns `-1` then, which `substring(...)` doesnt understand).

Reply:
but fileName always got a . + extension at the end cause of my extensionFilter at the beginning which filters these

Reply:
maybe try to give me an example file name + extension which you think could break

Reply:
> but fileName always got a .

no

Reply:
> > but fileName always got a .
> 
> no

give me an example pls

Reply:
> > > but fileName always got a .
> > 
> > 
> > no
> 
> give me an example pls

Dockerfile

Reply:
ur constantly resolving this, but the problem is still not fixed:
> this still has the problem of crashing if `fileName` has no `.` in it (`lastIndexOf` retuns `-1` then, which `substring(...)` doesnt understand).